crates/bevy_ecs/src/schedule/graph_utils.rs

```diff
@@ -274,3 +274,119 @@ where
         transitive_closure,
     }
 }
+
+/// Returns the simple cycles in a strongly-connected component of a directed graph.
+///
+/// The algorithm implemented comes from
+/// ["Finding all the elementary circuits of a directed graph"][1] by D. B. Johnson.
+///
+/// [1]: https://doi.org/10.1137/0204007
+pub fn simple_cycles_in_component<N>(graph: &DiGraphMap<N, ()>, scc: &[N]) -> Vec<Vec<N>>
+where
+    N: NodeTrait + Debug,
+{
+    let mut cycles = vec![];
+    let mut sccs = vec![scc.to_vec()];
+
+    while let Some(mut scc) = sccs.pop() {
```

Comment:
Why do we have a whole other Vec wrapping around `scc` as a Vec and then popping the `sccs` wrapper in a while-loop when there's always just this one Vec inside?

Reply:
Once a node is done being the "root", it's removed from the graph and we have to split the remaining subgraph into new SCCs. Those are pushed to `sccs`.

Reply:
Oh ok

Reply:
LOL I missed the last line yeah makes sense